packages/react/radio-group/src/RadioGroup.tsx

```diff
@@ -1,43 +1,46 @@
 import * as React from 'react';
 import { composeEventHandlers } from '@radix-ui/primitive';
+import { useLabelContext } from '@radix-ui/react-label';
 import { useComposedRefs } from '@radix-ui/react-compose-refs';
 import { createContext } from '@radix-ui/react-context';
+import { extendPrimitive } from '@radix-ui/react-primitive';
+import { RovingFocusGroup, RovingFocusItem } from '@radix-ui/react-roving-focus';
+import { Slot } from '@radix-ui/react-slot';
 import { useCallbackRef } from '@radix-ui/react-use-callback-ref';
 import { useControllableState } from '@radix-ui/react-use-controllable-state';
-import { Primitive, extendPrimitive } from '@radix-ui/react-primitive';
 import { Radio, RadioIndicator } from './Radio';
-import { useLabelContext } from '@radix-ui/react-label';
-import { RovingFocusGroup, useRovingFocus } from '@radix-ui/react-roving-focus';
 
 import type * as Polymorphic from '@radix-ui/react-polymorphic';
 
 /* -------------------------------------------------------------------------------------------------
  * RadioGroup
  * -----------------------------------------------------------------------------------------------*/
 const RADIO_GROUP_NAME = 'RadioGroup';
+const RADIO_GROUP_DEFAULT_TAG = 'div';
 
 type RadioGroupOwnProps = Polymorphic.Merge<
-  Polymorphic.OwnProps<typeof Primitive>,
+  Omit<
+    Polymorphic.OwnProps<typeof RovingFocusGroup>,
+    'currentTabStopId' | 'defaultCurrentTabStopId' | 'onCurrentTabStopIdChange' | 'onEntryFocus'
+  >,
   {
     name?: string;
     value?: string;
     defaultValue?: string;
     required?: React.ComponentProps<typeof Radio>['required'];
-    rovingFocus?: boolean;
```

Comment:
Removed all the dead code about handling no-roving case as it was a leftover from its use in `Toolbar` which has now been replaced by `ToggleGroup`.